test-scripts/src/main.rs

```diff
@@ -122,16 +136,62 @@ async fn main() -> anyhow::Result<()> {
         (params_out, args.out)
     };
 
+    let url = &args.base_url;
+    let make_url = move |path: &str| -> String {
+        let mut url = url.clone();
+        url.set_path(path);
+        url.to_string()
+    };
+
+    let candidates = {
+        if let Some(dir) = args.candidates_dir {
+            let dir = std::fs::read_dir(dir)?;
+            let mut candidates = Vec::new();
+            for entry in dir {
+                let entry = entry?;
+                let path = entry.path();
+                if let Some(ext) = path.extension() {
+                    if ext == "json" {
+                        let mut hasher = sha2::Sha256::new();
+                        let mut file = std::fs::File::open(&path)?;
+                        std::io::copy(&mut file, &mut hasher)?;
+                        let web_path = format!(
+                            "static/concordium/candidates/{}",
+                            path.file_name().context("No file name.")?.to_string_lossy()
+                        );
+                        candidates.push(ChecksumUrl {
+                            url:  make_url(&web_path),
+                            hash: contract::HashSha2256(hasher.finalize().into()),
+                        })
+                    }
+                }
+            }
+            candidates
+        } else {
+            (0..args.num_options)
+                .map(|c| {
+                    let mut url = args.base_url.clone();
+                    url.set_path(&c.to_string());
+                    contract::ChecksumUrl {
+                        url:  url.to_string(),
+                        hash: contract::HashSha2256([0u8; 32]),
+                    }
+                })
+                .collect()
+        }
```

Comment:
Obviously this will not suffice for the voting dApp as it checks the content at the url. Would it make sense to also write some valid JSON test data for each candidate to `args.out` for completeness and align the url written to the contract with the http server? I think this would provide a much clearer flow: that you as a user specify the `out` directory for the script and map the artifacts produced into the http server.

Reply:
I'm not sure which part you're commenting on.

The "if" part of this branch is the one intended for use with the voting dapp, and that puts the correct hashes in.

This else part was for testing the coordinator tool so I thought it's not necessary to have valid candidate URLs.

Do you still think I should add that?

Reply:
Yes my point is I think it would be usable to use this tool to create a set of candidates to be used as well to easily create all the data needed for an election, instead of having to go edit the example configuration to change the amount of candidates which can be selected from the voting dApp. It's not crucial but it's not hard to add either. What do you think?

Reply:
I have added this.

Reply:
The logic is that if you don't supply candidates-dir then candidates are created in out/candidates.